### PR TITLE
[Dashboard] Feature: Updated Mintable Modules

### PR DESCRIPTION
**Pull request by GWSzeto**
https://linear.app/thirdweb/project/[modular-contracts]-modules-ui-c7a2be28299d/overview

---

## PR-Codex overview
This PR focuses on improving the handling of mintable modules in the dashboard and thirdweb packages, specifically refining the conditions for checking roles and optimizing the URI handling for NFTs.

### Detailed summary
- Changed the condition in `module-instance.tsx` to check for "Mintable" instead of "Mintable-ignore".
- Updated URI handling in `mintWithRole.ts` for both `MintableERC721` and `MintableERC1155` to check if NFTs are not empty before processing.
- Refactored the use of `hasAllRoles` and `grantRoles` in `Mintable.tsx` for better role management.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by gregfromstl** [comment]
@GWSzeto is there an eng linear ticket for this? The one linked is under design and marked as completed. Sorry for being strict about this it just helps give context so we can review

File: packages/thirdweb/src/extensions/modules/MintableERC1155/mintWithRole.ts [hunk 1]
```diff
@@ -45,13 +45,17 @@ export function mintWithRole(
...
+      if (options.nft) {
+        const batchOfUris = await uploadOrExtractURIs(
+          [options.nft],
+          options.contract.client,
+        );
+        baseURI = getBaseUriFromBatch(batchOfUris);
+      }
 
       const tokenId = options.tokenId ?? maxUint256;
+      console.log("tokenId: ", tokenId);
```
Comment (MananTank) [inline packages/thirdweb/src/extensions/modules/MintableERC1155/mintWithRole.ts:58]: remove this log

**Review by MananTank — CHANGES_REQUESTED** [review]
please address comments

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/Mintable.tsx [hunk 2]
```diff
@@ -99,8 +99,8 @@ function MintableModule(props: ModuleInstanceProps) {
       if (!hasMinterRole.data) {
         const grantRoleTx = prepareContractCall({
           contract,
-          method: "function grantRole(address user, uint256 role) public",
-          params: [ownerAccount.address || "", MINTER_ROLE],
+          method: "function grantRoles(address user, uint256 roles)",
+          params: [ownerAccount.address, MINTER_ROLE],
```
[the remark this answers is not in the document]
Comment (GWSzeto) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/Mintable.tsx:103]: Updated to use the extension

**Review by MananTank — APPROVED** [review]

**Review by gregfromstl — APPROVED** [review]

**Comment by GWSzeto** [comment]
### Merge activity

* **Nov 19, 1:30 PM EST**: The merge label 'merge-queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=thirdweb-dev&repo=js) once it meets the requirements.
* **Nov 19, 1:30 PM EST**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=thirdweb-dev&repo=js).

**Pull request merged**